lib/ToolbarBadgeHub.jsm

```diff
@@ -243,10 +248,26 @@ class _ToolbarBadgeHub {
     }
     let toolbarbutton = document.getElementById(message.content.target);
     if (toolbarbutton) {
+      const badge = toolbarbutton.querySelector(".toolbarbutton-badge");
+      badge.classList.add("feature-callout");
       toolbarbutton.setAttribute("badged", true);
-      toolbarbutton
-        .querySelector(".toolbarbutton-badge")
-        .classList.add("feature-callout");
+      // If we have additional aria-label information for the notification
+      // we add this content to the hidden `toolbarbutton-text` node.
+      // We then use `aria-labelledby` to link this description to the button
+      // that received the notification badge.
+      if (message.content["aria-label"]) {
+        toolbarbutton.setAttribute(
+          "aria-labelledby",
+          "toolbarbutton-notification-description"
```

Comment:
As I understand it, #toolbarbutton-notification-description will just contain "New feature", not "What's New". Thus, I would've thought setting aria-labelledby like this would override "What's new" completely. However, you noted you'd tested this with VoiceOver and you saw both? That's good, but I'm a little concerned because I don't understand *how* that's working.

Does the Accessibility Inspector in the Browser Toolbox also show the concatenated name?

Reply:
From the Browser Toolbox:
<img width="381" alt="Screenshot 2019-09-30 at 11 39 54" src="https://user-images.githubusercontent.com/810040/65867636-4d9fd300-e366-11e9-8dc1-2bf792983462.png">

Reply:
[From MDN](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-labelledby_attribute#Possible_effects_on_user_agents_and_assistive_technology) it looks like `aria-labelledby` takes precedence over `aria-label` but this button only has a `label` defined.
If this behavior seems strange we could change it so that only the `labelledby` content is shown (moving `aria-label` responsibility completely on the badge, when active) and then revert when the badge is removed.